packages/openmrs/test/index.js

```diff
@@ -47,3 +59,239 @@ describe('execute', () => {
     });
   });
 });
+
+describe('request', () => {
+  it('should search for a patient', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah 1' },
+        method: 'GET',
+      })
+      .reply(200, { results: [{ display: 'Sarah 1' }] }, { ...jsonHeaders });
+    const state = { configuration };
+
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah 1' }
+    );
+
+    expect(body.results[0].display).to.eql('Sarah 1');
+  });
+  it('should auto-fetch patients with a limit', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 1' }],
+          links: [
+            {
+              rel: 'next',
+              uri: 'https://fn.openmrs.org/ws/rest/v1/patient?q=Sarah&limit=1&startIndex=1',
+              resourceAlias: null,
+            },
+          ],
+        },
+        { ...jsonHeaders }
+      );
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1, startIndex: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 2' }],
+        },
+        { ...jsonHeaders }
+      );
+
+    const state = { configuration };
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah', limit: 1 }
+    );
+    expect(body.results[0].display).to.eql('Sarah 1');
+    expect(body.results[1].display).to.eql('Sarah 2');
+  });
+
+  it('should not auto-fetch if the user sets startIdex', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1, startIndex: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 2' }],
+        },
+        { ...jsonHeaders }
+      );
+
+    const state = { configuration };
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah', limit: 1, startIndex: 1 }
+    );
+
+    expect(body.results[0].display).to.eql('Sarah 2');
+  });
+});
+
+describe('get', () => {
+  before(() => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/encounter/123',
+        method: 'GET',
+      })
+      .reply(200, { uuid: '123' }, { ...jsonHeaders });
+  });
+
+  it('should get an encounter by uuid', async () => {
+    const state = { configuration };
+
+    const { data } = await execute(get('encounter/123'))(state);
+
+    expect(data.uuid).to.eql('123');
+  });
+});
+
+describe('post', () => {
+  before(() => {
```

Comment:
I know there's only one test in this `describe`, but by creating this interceptor i the before block, you're sort of declaring that this interceptor will work for all these tests. It implies that the interceptor is shared.

Which is fine, and I agree with that, but you need to add `.persist()` or else the interceptor will be destroyed after the first request. It'll create confusion for the next dev coming in. The same for the other tests.

Reply:
Moved the testServer into it's own test because in each test the interceptor will be different so it doesn't make sense to add the mock in `before()`